docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -814,14 +814,34 @@ The extended `BeanConfigurator` accepts either a `io.quarkus.runtime.RuntimeValu
 [source,java]
 ----
 @BuildStep
-@Record(STATIC_INIT)
+@Record(STATIC_INIT) <1>
 SyntheticBeanBuildItem syntheticBean(TestRecorder recorder) {
    return SyntheticBeanBuildItem.configure(Foo.class).scope(Singleton.class)
+                .runtimeValue(recorder.createFoo()) <2>
+                .done();
+}
+----
+<1> By default, a synthetic bean is initialized during `STATIC_INIT`.
+<2> The bean instance is supplied by a value returned from a recorder method.
+
+It is possible to mark a synthetic bean to be initialized during `RUNTIME_INIT`:
+
+.`RUNTIME_INIT` `SyntheticBeanBuildItem` Example
+[source,java]
+----
+@BuildStep
+@Record(RUNTIME_INIT) <1>
+SyntheticBeanBuildItem syntheticBean(TestRecorder recorder) {
+   return SyntheticBeanBuildItem.configure(Foo.class).scope(Singleton.class)
+                .setRuntimeInit() <2>
                 .runtimeValue(recorder.createFoo())
                 .done();
 }
 ----
+<1> The recorder must be executed in the `ExecutionTime.RUNTIME_INIT` phase.
+<2> The bean instance is initialized during `RUNTIME_INIT`.
 
+IMPORTANT: Synthetic bean initialized during `RUNTIME_INIT` must not be accessed during `STATIC_INIT`. `RUNTIME_INIT` build steps that access a runtime-init synthetic bean should consume the `SyntheticBeansRuntimeInitBuildItem`.
```

Comment:
Can we have an example, or a link to some code to make this more tangible?

Reply:
Hm, I don't think we use it anywhere ATM. I will add an example...